examples-next/auth/schema.ts

```diff
@@ -0,0 +1,144 @@
+ * input for the relationship? Seems like it would have to be the second, but it's not clear from
+ * the documentation. Is this consistent with how the `gqlName` argument works?
+ *
+ * tbh I'm inclined to remove the gqlName argument unless there's a compelling reason to have it;
+ * esp. given it's marked as optional.
+ *
+ * Finally, what's the point of the `auth` operation? It seems to be related to the listAuth
+ * functionality that we've just replaced with the new auth & session funcationality, if so we can
+ * remove it from access control in the new interfaces.
+ *
+ * Based on the docs (see https://www.keystonejs.com/api/access-control#the-auth-operation) it
+ * doesn't look like we need it anymore.
+ */
+
+type GenericItem = { id: string | number; [path: string]: any };
+type Session = any; // we have this defined elsewhere
+type Input = Record<string, any>;
+
+type ListAccessInput = {
+  session?: Session;
+  listKey: string;
+} & (
+  | { operation: 'create'; input: Input }
+  | { operation: 'read'; item: GenericItem }
+  | { operation: 'update'; item: GenericItem; input: Input }
+  | { operation: 'delete'; item: GenericItem }
+);
+
+type FieldAccessInput = {
+  session?: Session;
+  listKey: string;
+  fieldPath: string;
+} & (
+  | { operation: 'create'; input: Input }
+  | { operation: 'read'; item: GenericItem }
+  | { operation: 'update'; item: GenericItem; input: Input }
+);
+
+export const lists = createSchema({
+  User: list({
+    access: {
+      // Only allow admins to delete users
+      delete: ({ session }) => session?.data?.isAdmin,
+    },
+    admin: {
+      // Since you can't delete users unless you're an admin, we hide the UI for it
+      hideDelete: ({ session }) => session?.data?.isAdmin,
+      listView: {
+        // These are the default columns that will be displayed in the list view
+        initialColumns: ['name', 'email', 'isAdmin'],
+      },
+    },
+    fields: {
+      // The user's name
+      name: text({ isRequired: true }),
+      // The user's email address, used as the identity field for auth
+      email: text({
+        isRequired: true,
+        isUnique: true,
+      }),
+      // The user's password, used as the secret field for auth
+      password: password({
+        access: {
+          // Passwords can always be set when creating items
+          // Users can change their own passwords, and Admins can change anyone's password
+          update: ({ session, item }) =>
+            session && (session.data.isAdmin || session.itemId === item.itemId),
+        },
+        admin: {
+          // Based on the same logic as update access, the password field is editable.
+          // The password field is hidden from non-Admin users (except for themselves)
+          itemView: {
+            fieldMode: ({ session, item }) =>
+              session && (session.data.isAdmin || session.itemId === item.itemId)
```

Comment:
Why is the id now `.itemId` instead of `.id`? Seems redundant, and doesn't match the GraphQL API's field name.

Reply:
Also a good catch! 👍